base/src/main/java/com/austindroids/austinfeedsme/common/events/EventsPresenter.kt

```diff
@@ -0,0 +1,150 @@
+package com.austindroids.austinfeedsme.common.events
+
+import android.text.TextUtils
+
+import com.austindroids.austinfeedsme.common.utils.DateUtils
+import com.austindroids.austinfeedsme.data.Event
+import com.austindroids.austinfeedsme.data.FilterableEventsRepository
+import com.austindroids.austinfeedsme.di.scopes.ActivityScoped
+import com.austindroids.austinfeedsme.events.EventsFilterType
+
+import java.util.ArrayList
+import java.util.Date
+import java.util.HashMap
+
+import javax.inject.Inject
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.functions.Consumer
+import io.reactivex.schedulers.Schedulers
+import timber.log.Timber
+
+import com.austindroids.austinfeedsme.data.Event.Type.BEER
+import com.austindroids.austinfeedsme.data.Event.Type.NONE
+import com.austindroids.austinfeedsme.data.Event.Type.PIZZA
+import com.austindroids.austinfeedsme.data.Event.Type.TACO
+
+@ActivityScoped
+class EventsPresenter @Inject
+constructor(private val repository: FilterableEventsRepository, private val view: EventsContract.View) : EventsContract.Presenter {
+
+    private val compositeDisposable = CompositeDisposable()
+
+    private var currentFiltering = EventsFilterType.ALL_EVENTS
+
+    override fun loadEvents() {
+        loadEvents(true)
+    }
+
+    private fun loadEvents(showProgress: Boolean) {
+        val eventDisposable = repository.getEventsRX(true, true)
+                .doOnSubscribe {
+                    if (showProgress) {
+                        view.showProgress()
+                    }
+                }.doOnError { throwable ->
+                    view.hideProgress()
+                    Timber.e(throwable)
+                }.subscribe(Consumer { events ->
+                    if (events.isEmpty()) {
+                        view.showNoEventsView()
+                        view.hideProgress()
+                        return@Consumer
+                    }
+
+                    val eventsToShow = filterEventsOnTimeSelection(events)
+
+                    view.hideProgress()
+                    view.showEvents(eventsToShow)
+                    setEventCounts(eventsToShow)
+                })
+
+        compositeDisposable.add(eventDisposable)
+    }
+
+
+    override fun refreshEvents() {
+        loadEvents(false)
+    }
+
+    override fun searchEvents(searchTerm: String) {
+        if (searchTerm.equals(SEARCH_RESET_STRING, ignoreCase = true)) {
+            loadEvents()
+            return
+        }
+
+        val lowerCaseSearch = searchTerm.toLowerCase()
+
+        val subscribe = repository.eventsRX
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ events ->
+                    val filteredEvents = events.filter {
```

Comment:
Consider explicitly naming this `it` as `event`, because it is multi-line lambda.